src/industry_cmd.cpp

```diff
@@ -115,12 +115,12 @@ IndustryType GetIndustryType(TileIndex tile)
  * This will ensure at once : proper access and
  * not allowing modifications of it.
  * @param thistype of industry (which is the index in _industry_specs)
- * @pre thistype < NUM_INDUSTRYTYPES
+ * @pre thistype <= NUM_INDUSTRYTYPES
  * @return a pointer to the corresponding industry spec
  */
 const IndustrySpec *GetIndustrySpec(IndustryType thistype)
 {
-	assert(thistype < NUM_INDUSTRYTYPES);
+	assert(thistype <= NUM_INDUSTRYTYPES);
 	return &_industry_specs[thistype];
```

Comment:
But if `thistype` is equal to `NUM_INDUSTRYTYPES` you can cause an invalid read, as `NUM_INDUSTRYTYPES` is out of the array bounds.